homeassistant/components/geocaching/entity.py

```diff
@@ -0,0 +1,16 @@
+"""Sensor entities for Geocaching."""
+
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .coordinator import GeocachingDataUpdateCoordinator
+
+
+# Base class for all platforms
+class GeocachingBaseEntity(CoordinatorEntity[GeocachingDataUpdateCoordinator]):
+    """Base class for Geocaching sensors."""
+
+    _attr_has_entity_name = True
+
+    def __init__(self, coordinator: GeocachingDataUpdateCoordinator) -> None:
+        """Initialize the Geocaching sensor."""
+        super().__init__(coordinator)
```

Comment:
Okay this is a good direction. Now 2 things:

1. Let's remove the constructor as it does nothing
2. Let's also create a GeocachingCacheEntity that inherits this class. This new class can have the device_info and `self.cache`, since you will be using it for more platforms